rally/challenges/default.json

```diff
@@ -0,0 +1,52 @@
+    {
+      "name": "append-no-conflicts",
+      "description": "Indexes the whole document corpus using Elasticsearch default settings. We only adjust the number of replicas as we benchmark a single node cluster and Rally will only start the benchmark if the cluster turns green. Document ids are unique so all index operations are append only.",
+      "default": true,
+      "schedule": [
+        {
+          "operation": "delete-index"
+        },
+        {
+          "operation": {
+            "operation-type": "create-index",
+            "settings": {{index_settings | default({}) | tojson}}
+          }
+        },
+        {
+          "name": "check-cluster-health",
+          "operation": {
+            "operation-type": "cluster-health",
+            "index": "logs-*",
+            "request-params": {
+              "wait_for_status": "{{cluster_health | default('green')}}",
+              "wait_for_no_relocating_shards": "true"
+            }
+          }
+        },
+        {
+          "name": "index-apm-error",
```

Comment:
You are bulk-indexing one index after the other so you might be prone to ordering bias (i.e. the results would be different if you change the order of those three operations).

Another problem is that this does not do what you intend to do. The parameter to specify the target indices is called `indices` instead of `index` *and* it needs to be specified on the operation itself (see [reference docs](https://esrally.readthedocs.io/en/stable/track.html#bulk)). I also suggest you just inline the operation. E.g.:

```json
{
  "name": "index-apm-transaction",
  "operation": {
    "operation-type": "bulk",
    "indices": ["apm-6.4.0-transaction"],
    "bulk-size": {{bulk_size | default(1000)}},
    "ingest-percentage": {{ingest_percentage | default(100)}}
  },
  "warmup-time-period": 120,
  "clients": {{bulk_indexing_clients | default(8)}}
}
```

W.r.t. to the `indices` parameter you also need to be specific because Rally tries to match the specified indices against the target indices specified in the `corpora` block. I guess the reason you have specified a `*` is that you want it to match 6.4.0 and 7.0.0 indices. This is *usually* solved a bit differently (by creating a track repository and using branches) but you could also use a track parameter instead and use Rally's template language support to replace the track parameter for you. E.g. you'd specify `--track-param="es_version:6.4.0"` on the command line and instead of specifying `"indices": ["apm-6.4.0-transaction"]` you can also specify `"indices": ["apm-{{es_version}}-transaction"]`.